tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptionstatus/SubscriptionStatusControllerDelayedCancellationTest.java

```diff
@@ -0,0 +1,109 @@
+        ZonedDateTime timestamp = ZonedDateTime.now().minusSeconds(5);
+        String message = "Unsatisfied with the service";
+        String reasonCode = "CHURN";
+
+        // when
+        CancellationRequest request = new CancellationRequest(
+                new CancellationOptions.Builder()
+                        .cancellationMessage(message)
+                        .reasonCode(reasonCode)
+                        .build()
+        );
+        DelayedCancellationResponse delayedCancellationResponse = subscriptionStatusController
+                .initiateDelayedCancellation(subscription.getId(), request);
+        Subscription updatedSubscription = subscriptionsController.readSubscription(subscription.getId(), null)
+                .getSubscription();
+
+        // then
+        assertThat(delayedCancellationResponse.getMessage())
+                .isEqualTo("This subscription is scheduled to be canceled at the end of the current period");
+
+        assertThat(delayedCancellationResponse.getAdditionalProperties()).isEmpty();
+        assertThat(updatedSubscription).usingRecursiveComparison()
+                .ignoringFields("updatedAt", "cancelAtEndOfPeriod", "cancellationMessage",
+                        "currentBillingAmountInCents", "reasonCode", "scheduledCancellationAt", "delayedCancelAt")
+                .isEqualTo(subscription);
+        assertThat(updatedSubscription.getUpdatedAt()).isAfter(timestamp);
+        assertThat(updatedSubscription.getScheduledCancellationAt()).isAfter(timestamp);
+        assertThat(updatedSubscription.getDelayedCancelAt()).isAfter(timestamp);
+        assertThat(updatedSubscription.getCancelAtEndOfPeriod()).isTrue();
+        assertThat(updatedSubscription.getReasonCode()).isEqualTo(reasonCode);
+        assertThat(updatedSubscription.getCancellationMessage()).isEqualTo(message);
+    }
+
+    @Test
+    void shouldNotCancelNonExistentSubscription() {
+        assertNotFound(() -> subscriptionStatusController.initiateDelayedCancellation(99999999, null));
+    }
+
+    @Test
+    void shouldNotCancelAlreadyCanceledSubscriptionExceedingMessagesLimits() throws IOException, ApiException {
+        String reasonCode = "a".repeat(256);
+        String cancellationMessage = "a".repeat(65536);
+        CancellationRequest cancellationRequest = new CancellationRequest(
+                new CancellationOptions.Builder()
+                        .cancellationMessage(cancellationMessage)
+                        .reasonCode(reasonCode)
+                        .build()
+        );
+
+        // when - then
+        new ApiExceptionAssert(() -> subscriptionStatusController.initiateDelayedCancellation(subscriptionForErrors.getId(), cancellationRequest))
+                .hasErrorCode(422)
+                .hasMessage("HTTP Response Not OK. Status code: 422. Response: '{errors:[reason_code size cannot be greater than 255,cancellation_message size cannot be greater than 65535]}'.");
```

Comment:
Is this ErrorListResponse (or something similar?) - then you can use specific assertion with hasMessages method to check if model is properly deserialized.

Reply:
right, missed this test case